In [1]:
!pip install -q -U google-generativeai

In [2]:
import google.generativeai as genai 

/home/manish/anaconda3/envs/cuda121/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [ ]:
google_api = 'AIzaSyAMlYqwvuQgekl8nlqc56XTqJVBufszrBU'
#  AIzaSyAMlYqwvuQgekl8nlqc56XTqJVBufszrBU
genai.configure(api_key=google_api)

In [5]:
for m in genai.list_models():
    if 'generateContent' in m.supported_generation_methods:
        print(m.name)

models/gemini-1.5-pro-latest
models/gemini-1.5-pro-002
models/gemini-1.5-pro
models/gemini-1.5-flash-latest
models/gemini-1.5-flash
models/gemini-1.5-flash-002
models/gemini-1.5-flash-8b
models/gemini-1.5-flash-8b-001
models/gemini-1.5-flash-8b-latest
models/gemini-2.5-pro-preview-03-25
models/gemini-2.5-flash-preview-05-20
models/gemini-2.5-flash
models/gemini-2.5-flash-lite-preview-06-17
models/gemini-2.5-pro-preview-05-06
models/gemini-2.5-pro-preview-06-05
models/gemini-2.5-pro
models/gemini-2.0-flash-exp
models/gemini-2.0-flash
models/gemini-2.0-flash-001
models/gemini-2.0-flash-exp-image-generation
models/gemini-2.0-flash-lite-001
models/gemini-2.0-flash-lite
models/gemini-2.0-flash-preview-image-generation
models/gemini-2.0-flash-lite-preview-02-05
models/gemini-2.0-flash-lite-preview
models/gemini-2.0-pro-exp
models/gemini-2.0-pro-exp-02-05
models/gemini-exp-1206
models/gemini-2.0-flash-thinking-exp-01-21
models/gemini-2.0-flash-thinking-exp
models/gemini-2.0-flash-thinking-exp

In [6]:

# Model Configuration
MODEL_CONFIG = {
  "temperature": 0.2,
  "top_p": 1,
  "top_k": 32,
  "max_output_tokens": 4096,
}

## Safety Settings of Model
safety_settings = [
  {
    "category": "HARM_CATEGORY_HARASSMENT",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  },
  {
    "category": "HARM_CATEGORY_HATE_SPEECH",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  },
  {
    "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  },
  {
    "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  }
]

In [59]:
model = genai.GenerativeModel(
                            model_name='gemini-2.5-flash',
                            # model_name='gemini-2.5-pro',
                              generation_config=MODEL_CONFIG,
                              safety_settings=safety_settings)
model

genai.GenerativeModel(
    model_name='models/gemini-2.5-flash',
    generation_config={'temperature': 0.2, 'top_p': 1, 'top_k': 32, 'max_output_tokens': 4096},
    safety_settings={<HarmCategory.HARM_CATEGORY_HARASSMENT: 7>: <HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE: 2>, <HarmCategory.HARM_CATEGORY_HATE_SPEECH: 8>: <HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE: 2>, <HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: 9>: <HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE: 2>, <HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: 10>: <HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE: 2>},
    tools=None,
    system_instruction=None,
    cached_content=None
)

In [60]:
from pathlib import Path

def image_format(image_path):
    img = Path(image_path)

    if not img.exists():
        raise FileNotFoundError(f"Could not find image: {img}")

    image_parts = [
        {
            "mime_type": "image/png", ## Mime type are PNG - image/png. JPEG - image/jpeg. WEBP - image/webp
            "data": img.read_bytes()
        }
    ]
    return image_parts


image_format('/home/manish/Desktop/projects/etfp/bill_image_receipt.png')

[{'mime_type': 'image/png',
  'data': b'\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x02\x83\x00\x00\x04M\x08\x02\x00\x00\x00\xa6>\xdd\'\x00\x00J\xaeIDATx\xda\xed\xdd\xe1\x92\xa3:\xb2\xa8\xd1~\xff\x97\xe6F\x9c\x1b11\xd3]\x05\xa9\xcc\x94\x10\xb0\xd6\xaf\x99\xde.\x8c1\xe8\x03\x8c\xf1\x9f\x03\x00\xb8\xcf\x1f\x8b\x00\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\xd9xu\xa99\x9f\xda\xb3^\xbb\x95a\xea\x9a\xe3\xbd@\x89A\x89\x8d\xfe\xf7\xaf9\xde\x0b\x94\x18\x94\xd8\xe8\x7f\xf3\x9a\xe3\xbd@\x89A\x89\x8d\xfe7\xaf9\xde\x0b\x94\x

In [61]:
def gemini_output(image_path,
                  system_prompt,
                  user_prompt):
    
    image_info = image_format(image_path)
    print(image_info[0])
    input_prompt = [system_prompt, image_info[0], user_prompt]
    response = model.generate_content(input_prompt)

    return response.text




In [62]:
system_prompt = """
               You are a specialist in comprehending receipts.
               Input images in the form of receipts will be provided to you,
               and your task is to respond to questions based on the content of the input image.
               """

image_path = "/home/manish/Desktop/projects/etfp/bill_image_receipt.png"

user_prompt = "What is the balance amount in the image?"

gemini_output(image_path, system_prompt, user_prompt)

{'mime_type': 'image/png', 'data': b'\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x02\x83\x00\x00\x04M\x08\x02\x00\x00\x00\xa6>\xdd\'\x00\x00J\xaeIDATx\xda\xed\xdd\xe1\x92\xa3:\xb2\xa8\xd1~\xff\x97\xe6F\x9c\x1b11\xd3]\x05\xa9\xcc\x94\x10\xb0\xd6\xaf\x99\xde.\x8c1\xe8\x03\x8c\xf1\x9f\x03\x00\xb8\xcf\x1f\x8b\x00\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\x01@\x89\x01\x00%\x06\x00%\x06\x00\x94\x18\x00\x94\x18\x00\x94\x18\x00Pb\x00Pb\x00@\x89\xd9xu\xa99\x9f\xda\xb3^\xbb\x95a\xea\x9a\xe3\xbd@\x89A\x89\x8d\xfe\xf7\xaf9\xde\x0b\x94\x18\x94\xd8\xe8\x7f\xf3\x9a\xe3\xbd@\x89A\x89\x8d\xfe7\xaf9\xde\x0b\x94\x18f

'The Grand Total, which is the balance amount before payment, is **376.40**.'

In [17]:
system_prompt = """
               You are a specialist in comprehending receipts.
               Input images in the form of receipts will be provided to you,
               and your task is to respond to questions based on the content of the input image.
               """
#system_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image "
# image_path = "/content/handwritten.png"

user_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image "
     

In [18]:
output = gemini_output(image_path, system_prompt, user_prompt)
output

'```json\n{\n  "receipt_details": {\n    "business_info": {\n      "name": "The Lone Pine",\n      "address": "43 Manchester Road, 12480 Brisbane, Australia",\n      "phone": "617-3236-6207"\n    },\n    "transaction_info": {\n      "invoice_number": "08000008",\n      "table_number": "25",\n      "date": "09/04/08",\n      "time": "12:45"\n    },\n    "items": [\n      {\n        "item": "Carlsberg Bottle",\n        "quantity": 2,\n        "price": 16.00\n      },\n      {\n        "item": "Heineken Draft Standard",\n        "quantity": 3,\n        "price": 24.60\n      },\n      {\n        "item": "Heineken Draft Half Liter",\n        "quantity": 1,\n        "price": 15.20\n      },\n      {\n        "item": "Carlsberg Bucket (5 bottles)",\n        "quantity": 2,\n        "price": 80.00\n      },\n      {\n        "item": "Grilled Chicken Breast",\n        "quantity": 4,\n        "price": 74.00\n      },\n      {\n        "item": "Sirloin Steak",\n        "quantity": 3,\n        "pri

In [19]:
from IPython.display import Markdown
Markdown(output)

```json
{
  "receipt_details": {
    "business_info": {
      "name": "The Lone Pine",
      "address": "43 Manchester Road, 12480 Brisbane, Australia",
      "phone": "617-3236-6207"
    },
    "transaction_info": {
      "invoice_number": "08000008",
      "table_number": "25",
      "date": "09/04/08",
      "time": "12:45"
    },
    "items": [
      {
        "item": "Carlsberg Bottle",
        "quantity": 2,
        "price": 16.00
      },
      {
        "item": "Heineken Draft Standard",
        "quantity": 3,
        "price": 24.60
      },
      {
        "item": "Heineken Draft Half Liter",
        "quantity": 1,
        "price": 15.20
      },
      {
        "item": "Carlsberg Bucket (5 bottles)",
        "quantity": 2,
        "price": 80.00
      },
      {
        "item": "Grilled Chicken Breast",
        "quantity": 4,
        "price": 74.00
      },
      {
        "item": "Sirloin Steak",
        "quantity": 3,
        "price": 96.00
      },
      {
        "item": "Coke",
        "quantity": 1,
        "price": 3.50
      },
      {
        "item": "Ice Cream",
        "quantity": 5,
        "price": 18.00
      }
    ],
    "summary": {
      "subtotal": 327.30,
      "sales_gov_tax": {
        "percentage": "5%",
        "amount": 16.36
      },
      "service_charge": {
        "percentage": "10%",
        "amount": 32.73
      },
      "grand_total": 376.40
    },
    "payment_info": {
      "method": "Cash",
      "amount_paid": 400.00,
      "change": 23.60
    },
    "footer_messages": [
      "Thank you and see you again!",
      "John",
      "Bring this bill back within the next 10 days and get 15% discount on that day's food bill..."
    ]
  }
}
```

---

In [ ]:
system_prompt = """
               You are a specialist in comprehending receipts.
               Input images in the form of receipts will be provided to you,
               and your task is to respond to questions based on the content of the input image.
               """
#system_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image "
image_path = "/home/manish/Desktop/projects/etfp/temp/image_19.png"

user_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image "

output = gemini_output(image_path, system_prompt, user_prompt)
     

from IPython.display import Markdown
output = Markdown(output)
output

```json
{
  "invoice_header": {
    "invoice_no": "12",
    "dated": "6-May-2021",
    "page": "3"
  },
  "biller_details": {
    "name": "High Furnace Manufacturer",
    "address": "Unit 1, Sector 2, Khanpur road Jaisalmer, Rajasthan.",
    "gst": "23AM2359012D1ZW"
  },
  "consignee_details": {
    "name": "JK Energies Limited",
    "address": "House No. 1, Kolar road, Rajgarh, Madhya Pradesh.",
    "gst": "23KM4589012B1ZS"
  },
  "shipping_delivery_details": {
    "delivery_note": null,
    "mode_terms_of_payment": null,
    "supplier_ref": "12",
    "other_reference": null,
    "buyer_order_no": null,
    "buyer_order_dated": null,
    "despatch_document_no": null,
    "delivery_note_date": null,
    "despatched_through": null,
    "destination": "RAJGARH",
    "bill_of_lading_rr_no": null,
    "motor_vehicle_no": null,
    "terms_of_delivery": null
  },
  "item_summary": [
    {
      "description": "SGST R/OFF",
      "amount": "39,521.30"
    },
    {
      "description": "Less",
      "amount": "-0.03"
    }
  ],
  "financial_summary": {
    "total_amount_chargeable": {
      "value": "10,43,481.00",
      "in_words": "INR Ten Lakh Forty Three Thousand Four Hundred Eighty One Only"
    },
    "hsn_sac_breakdown": [
      {
        "hsn_sac": "85414012",
        "taxable_value": "4,86,641.28",
        "central_tax": {
          "rate": "2.50%",
          "amount": "12,166.02"
        },
        "state_tax": {
          "rate": "2.50%",
          "amount": "12,166.02"
        },
        "total_tax_amount": "24,332.04"
      },
      {
        "hsn_sac": "730890",
        "taxable_value": "1,62,304.77",
        "central_tax": {
          "rate": "9%",
          "amount": "14,607.43"
        },
        "state_tax": {
          "rate": "9%",
          "amount": "14,607.43"
        },
        "total_tax_amount": "29,214.86"
      },
      {
        "hsn_sac": "841391",
        "taxable_value": "74,777.69",
        "central_tax": {
          "rate": "9%",
          "amount": "6,729.99"
        },
        "state_tax": {
          "rate": "9%",
          "amount": "6,729.99"
        },
        "total_tax_amount": "13,459.98"
      },
      {
        "hsn_sac": "84137010",
        "taxable_value": "2,40,714.69",
        "central_tax": {
          "rate": "2.50%",
          "amount": "6,017.86"
        },
        "state_tax": {
          "rate": "2.50%",
          "amount": "6,017.86"
        },
        "total_tax_amount": "12,035.72"
      }
    ],
    "hsn_sac_breakdown_totals": {
      "total_taxable_value": "9,64,438.43",
      "total_central_tax_amount": "39,521.30",
      "total_state_tax_amount": "39,521.30",
      "total_tax_amount": "79,042.60"
    },
    "tax_amount_in_words": "INR Seventy Nine Thousand Forty Two and Sixty paise Only"
  },
  "company_bank_details": {
    "bank_name": "IDFC FIRST BANK",
    "account_no": null,
    "branch_ifs_code": null
  },
  "company_pan": null,
  "declaration": "We declare that this invoice shows the actual price of the goods described and that all particulars are true and correct.",
  "jurisdiction": "SUBJECT TO JAIPUR JURISDICTION",
  "note": "This is a Computer Generated Invoice"
}
```

---

In [22]:
import google.generativeai as genai 

google_api = 'AIzaSyAMlYqwvuQgekl8nlqc56XTqJVBufszrBU'
genai.configure(api_key=google_api)

# Model Configuration
MODEL_CONFIG = {
  "temperature": 0.2,
  "top_p": 1,
  "top_k": 32,
  "max_output_tokens": 4096,
}

## Safety Settings of Model
safety_settings = [
  {
    "category": "HARM_CATEGORY_HARASSMENT",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  },
  {
    "category": "HARM_CATEGORY_HATE_SPEECH",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  },
  {
    "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  },
  {
    "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
    "threshold": "BLOCK_MEDIUM_AND_ABOVE"
  }
]

model = genai.GenerativeModel(
                            model_name='gemini-2.5-flash',  # 6 min takes to 4 image 
                            # model_name='gemini-2.5-pro',
                              generation_config=MODEL_CONFIG,
                              safety_settings=safety_settings)
model



def gemini_output(image_path,
                  system_prompt,
                  user_prompt):
    
    
    input_prompt = [system_prompt, image_path, user_prompt]
    response = model.generate_content(input_prompt)

    return response.text




In [23]:
def process_multiple_images(image_paths,
                            system_prompt,
                            user_prompt):

    all_results = []

    for image_path in image_paths:
        result = gemini_output(image_path, system_prompt, user_prompt)
        all_results.append({
            result
        })

    return all_results
    



In [24]:
system_prompt = """
               You are a specialist in comprehending receipts.
               Input images in the form of receipts will be provided to you,
               and your task is to respond to questions based on the content of the input image.
               """

image_paths = [
        "/home/manish/Desktop/projects/etfp/temp/image_19.png",
        "/home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png",
        # "/home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg",
        # "/home/manish/Desktop/projects/etfp/temp/image_20.png"
]

user_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image "

output = process_multiple_images(image_paths, system_prompt, user_prompt)
output


[{'I understand you\'ve provided a file path `/home/manish/Desktop/projects/etfp/temp/image_19.png` for the receipt image. However, as an AI, I cannot directly access local file paths on your computer or browse your desktop.\n\nTo help you, please provide the content of the receipt image. You can do this by:\n\n1.  **Uploading the image directly** if the interface allows it.\n2.  **Describing the key information** from the invoice in detail (e.g., "Invoice number: INV-2023-001, Date: 2023-10-26, Vendor: ABC Store, Address: 123 Main St, Item 1: Coffee, Qty: 2, Price: 3.50, etc.").\n\nOnce I have the image content, I will convert the invoice data into the requested JSON format.'},
 {"I understand you'd like me to process the receipt image located at `/home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png` and convert its data into JSON format.\n\nHowever, as an AI, I cannot directly access local files on your computer or view images from a file path like the one you provided. My 

[{'I understand you\'ve provided a file path `/home/manish/Desktop/projects/etfp/temp/image_19.png` for the receipt image. However, as an AI, I cannot directly access local file paths on your computer or browse your desktop.\n\nTo help you, please provide the content of the receipt image. You can do this by:\n\n1.  **Uploading the image directly** if the interface allows it.\n2.  **Describing the key information** from the invoice in detail (e.g., "Invoice number: INV-2023-001, Date: 2023-10-26, Vendor: ABC Store, Address: 123 Main St, Item 1: Coffee, Qty: 2, Price: 3.50, etc.").\n\nOnce I have the image content, I will convert the invoice data into the requested JSON format.'},
 {"I understand you'd like me to process the receipt image located at `/home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png` and convert its data into JSON format.\n\nHowever, as an AI, I cannot directly access local files on your computer or view images from a file path like the one you provided. My capabilities are limited to processing text inputs.\n\nTo help you, please provide the text content of the receipt (e.g., by transcribing it, or describing the key details like store name, date, items, quantities, prices, subtotal, tax, total, etc.). Once you provide the text, I will gladly convert it into the requested JSON format."}]

----

---

#### 4 image takes 5 min

In [29]:
import base64
import google.generativeai as genai
from IPython.display import Markdown

# Configure Gemini (make sure you have your API key set up)
genai.configure(api_key="AIzaSyAMlYqwvuQgekl8nlqc56XTqJVBufszrBU")

def encode_image(image_path):
    """Encode image to base64"""
    with open(image_path, "rb") as image_file:
        return base64.b64encode(image_file.read()).decode('utf-8')

def process_single_image(image_path, system_prompt, user_prompt):
    """Process a single image with Gemini"""
    # Encode the image
    base64_image = encode_image(image_path)
    
    # Create the model
    model = genai.GenerativeModel('gemini-2.5-flash')
    
    # Prepare the content
    image_part = {
        "mime_type": "image/png",  # or "image/jpeg" depending on your image
        "data": base64_image
    }
    
    # Generate content
    response = model.generate_content([
        system_prompt,
        image_part,
        user_prompt
    ])
    
    return response.text

def process_multiple_images(image_paths, system_prompt, user_prompt):
    """Process multiple images and combine results"""
    all_results = []
    
    for image_path in image_paths:
        print(f"Processing: {image_path}")
        try:
            result = process_single_image(image_path, system_prompt, user_prompt)
            all_results.append({
                "image_path": image_path,
                "result": result
            })
        except Exception as e:
            print(f"Error processing {image_path}: {e}")
            all_results.append({
                "image_path": image_path,
                "error": str(e)
            })
    
    return all_results

# Your prompts
system_prompt = """You are a specialist in comprehending receipts.
Input images in the form of receipts will be provided to you,
and your task is to respond to questions based on the content of the input image."""

user_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image"

image_paths = [
    "/home/manish/Desktop/projects/etfp/temp/image_19.png",
    "/home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png",
    "/home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg",
    "/home/manish/Desktop/projects/etfp/temp/image_20.png"
]

# Process all images
results = process_multiple_images(image_paths, system_prompt, user_prompt)

# Display results
for result in results:
    print(f"\n=== Results for {result['image_path']} ===")
    if 'result' in result:
        display(Markdown(result['result']))
    else:
        print(f"Error: {result['error']}")

Processing: /home/manish/Desktop/projects/etfp/temp/image_19.png
Processing: /home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png
Processing: /home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg
Processing: /home/manish/Desktop/projects/etfp/temp/image_20.png

=== Results for /home/manish/Desktop/projects/etfp/temp/image_19.png ===


```json
{
  "invoice_details": {
    "invoice_number": "12",
    "invoice_date": "6-May-2021",
    "delivery_note": null,
    "mode_terms_of_payment": null,
    "supplier_ref": "12",
    "other_references": null,
    "buyer_order_no": null,
    "buyer_order_date": null,
    "despatch_document_no": null,
    "delivery_note_date": null,
    "despatched_through": null,
    "destination": "RAJGARH",
    "bill_of_lading_rr_no": null,
    "motor_vehicle_no": null,
    "terms_of_delivery": null
  },
  "billers_details": {
    "name": "High Furnace Manufacturer",
    "address": "Unit 1, Sector 2, Khanpur road Jaisalmer, Rajasthan.",
    "gstin": "23AM2359012D1ZW"
  },
  "consignee_details": {
    "name": "JK Energies Limited",
    "address": "House No. 1, Kolar road, Rajgarh, Madhya Pradesh.",
    "gstin": "23KM4589012B1ZS"
  },
  "item_summary_section": {
    "description_of_goods": [],
    "sgst_r_off": {
      "amount": "39,521.30",
      "adjustment": "(-)-0.03"
    },
    "total_amount_from_section": "10,43,481.00"
  },
  "hsn_sac_wise_summary": [
    {
      "hsn_sac": "85414012",
      "taxable_value": "4,86,641.28",
      "central_tax": {
        "rate": "2.50%",
        "amount": "12,166.02"
      },
      "state_tax": {
        "rate": "2.50%",
        "amount": "12,166.02"
      },
      "total_tax_amount": "24,332.04"
    },
    {
      "hsn_sac": "730890",
      "taxable_value": "1,62,304.77",
      "central_tax": {
        "rate": "9%",
        "amount": "14,607.43"
      },
      "state_tax": {
        "rate": "9%",
        "amount": "14,607.43"
      },
      "total_tax_amount": "29,214.86"
    },
    {
      "hsn_sac": "841391",
      "taxable_value": "74,777.69",
      "central_tax": {
        "rate": "9%",
        "amount": "6,729.99"
      },
      "state_tax": {
        "rate": "9%",
        "amount": "6,729.99"
      },
      "total_tax_amount": "13,459.98"
    },
    {
      "hsn_sac": "84137010",
      "taxable_value": "2,40,714.69",
      "central_tax": {
        "rate": "2.50%",
        "amount": "6,017.86"
      },
      "state_tax": {
        "rate": "2.50%",
        "amount": "6,017.86"
      },
      "total_tax_amount": "12,035.72"
    }
  ],
  "totals": {
    "total_taxable_value": "9,64,438.43",
    "total_central_tax_amount": "39,521.30",
    "total_state_tax_amount": "39,521.30",
    "total_tax_amount": "79,042.60",
    "amount_chargeable_in_words": "INR Ten Lakh Forty Three Thousand Four Hundred Eighty One Only",
    "tax_amount_in_words": "INR Seventy Nine Thousand Forty Two and Sixty paise Only"
  },
  "company_bank_details": {
    "bank_name": "IDFC FIRST BANK",
    "account_no": null,
    "branch_ifs_code": null
  },
  "company_pan": null,
  "declaration": "We declare that this invoice shows the actual price of the goods described and that all particulars are true and correct.",
  "jurisdiction": "SUBJECT TO JAIPUR JURISDICTION",
  "document_type": "This is a Computer Generated Invoice"
}
```


=== Results for /home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png ===


```json
{
  "receipt_details": {
    "restaurant_name": "The Lone Pine",
    "address": "43 Manchester Road, 12480 Brisbane, Australia",
    "phone_number": "617-3236-6207",
    "invoice_number": "08000008",
    "table_number": "25",
    "date": "09/04/08",
    "time": "12:45",
    "items": [
      {
        "quantity": 2,
        "description": "Carlsberg Bottle",
        "amount": 16.00
      },
      {
        "quantity": 3,
        "description": "Heineken Draft Standard",
        "amount": 24.60
      },
      {
        "quantity": 1,
        "description": "Heineken Draft Half Liter",
        "amount": 15.20
      },
      {
        "quantity": 2,
        "description": "Carlsberg Bucket (5 bottles)",
        "amount": 80.00
      },
      {
        "quantity": 4,
        "description": "Grilled Chicken Breast",
        "amount": 74.00
      },
      {
        "quantity": 3,
        "description": "Sirloin Steak",
        "amount": 96.00
      },
      {
        "quantity": 1,
        "description": "Coke",
        "amount": 3.50
      },
      {
        "quantity": 5,
        "description": "Ice Cream",
        "amount": 18.00
      }
    ],
    "subtotal": 327.30,
    "taxes": [
      {
        "type": "Sales/Gov Tax",
        "percentage": "5%",
        "amount": 16.36
      }
    ],
    "charges": [
      {
        "type": "Service Charge",
        "percentage": "10%",
        "amount": 32.73
      }
    ],
    "grand_total": 376.40,
    "payment_details": {
      "type": "Cash",
      "amount_paid": 400.00,
      "change_due": 23.60
    },
    "cashier_or_server": "John",
    "notes": "Bring this bill back within the next 10 days and get 15% discount on that day's food bill..."
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg ===


```json
{
  "merchant_name": "VISHAL PUNJABI VAISHNO BHOJANALYA",
  "address": "HOTEL SUN-N-SUN, BOULEVARD ROAD, G-2, SRINAGAR",
  "date": "22/05/07",
  "invoice_number": "1687",
  "customer_name": "Cash.",
  "items": [
    {
      "quantity": 1,
      "description": "Khichri with Butter",
      "amount": 35.00
    },
    {
      "quantity": 1,
      "description": "Khoya Kaju",
      "amount": 50.00
    },
    {
      "quantity": 2,
      "description": "Curd",
      "amount": 24.00
    },
    {
      "quantity": 4,
      "description": "Rati",
      "amount": 12.00
    },
    {
      "quantity": 1,
      "description": "Rice",
      "amount": 25.00
    }
  ],
  "total_amount": 146.00,
  "currency": "Rs."
}
```


=== Results for /home/manish/Desktop/projects/etfp/temp/image_20.png ===


```json
{
  "invoice_details": {
    "invoice_no": "12",
    "invoice_date": "6-May-2021",
    "delivery_note": null,
    "mode_terms_of_payment": null,
    "supplier_ref": "12",
    "other_reference": null,
    "buyer_order_no": null,
    "buyer_order_date": null,
    "despatch_document_no": null,
    "delivery_note_date": null,
    "despatched_through": null,
    "destination": "RAJGARH",
    "bill_of_lading_rr_no": null,
    "motor_vehicle_no": null,
    "terms_of_delivery": null
  },
  "biller_details": {
    "name": "High Furnace Manufacturer",
    "address": "Unit 1, Sector 2, Khanpur road Jaisalmer, Rajasthan.",
    "gst": "23AM2359012D1ZW"
  },
  "consignee_details": {
    "name": "JK Energies Limited",
    "address": "House No. 1, Kolar road, Rajgarh, Madhya Pradesh.",
    "gst": "23KM4589012B1ZS"
  },
  "line_items": [
    {
      "sl_no": 14,
      "description_of_goods": "DC Submersible Pumps 3 HP 30 MTR Head",
      "hsn_sac": "84137010",
      "gst_rate": "5 %",
      "quantity": 1,
      "unit": "Nos.",
      "rate_per_unit": 36934.54,
      "amount": 36934.54
    },
    {
      "sl_no": 15,
      "description_of_goods": "SPV MODULES 24V 335W POLY",
      "hsn_sac": "85414012",
      "gst_rate": "5 %",
      "quantity": 9,
      "unit": "Nos.",
      "rate_per_unit": 6554.59,
      "amount": 58991.31
    },
    {
      "sl_no": 16,
      "description_of_goods": "3 HP Capacity Solar Iron Structure",
      "hsn_sac": "730890",
      "gst_rate": "18 %",
      "quantity": 1,
      "unit": "SET",
      "rate_per_unit": 18353.39,
      "amount": 18353.39
    },
    {
      "sl_no": 17,
      "description_of_goods": "BOS 3 HP PSMP @ 70 MTR",
      "hsn_sac": "841391",
      "gst_rate": "18 %",
      "quantity": 1,
      "unit": "SET",
      "rate_per_unit": 11834.51,
      "amount": 11834.51
    },
    {
      "sl_no": 18,
      "description_of_goods": "DC Submersible Pumps 3 HP 70 MTR Head",
      "hsn_sac": "84137010",
      "gst_rate": "5 %",
      "quantity": 1,
      "unit": "Nos.",
      "rate_per_unit": 36376.56,
      "amount": 36376.56
    },
    {
      "sl_no": 19,
      "description_of_goods": "SPV MODULES 24V 320W POLY",
      "hsn_sac": "85414012",
      "gst_rate": "5 %",
      "quantity": 15,
      "unit": "Nos.",
      "rate_per_unit": 6048.99,
      "amount": 90734.85
    },
    {
      "sl_no": 20,
      "description_of_goods": "3 HP Capacity Solar Iron Structure",
      "hsn_sac": "730890",
      "gst_rate": "18 %",
      "quantity": 1,
      "unit": "SET",
      "rate_per_unit": 18178.98,
      "amount": 18178.98
    },
    {
      "sl_no": 21,
      "description_of_goods": "2 HP Capacity Solar Iron Structure",
      "hsn_sac": "730890",
      "gst_rate": "18 %",
      "quantity": 1,
      "unit": "SET",
      "rate_per_unit": 12740.00,
      "amount": 12740.00
    },
    {
      "sl_no": 22,
      "description_of_goods": "BOS 5 HP PSMP @ 100 MTR",
      "hsn_sac": "841391",
      "gst_rate": "18 %",
      "quantity": 1,
      "unit": "SET",
      "rate_per_unit": 18722.68,
      "amount": 18722.68
    },
    {
      "sl_no": 23,
      "description_of_goods": "DC Submersible Pumps 5 HP 100 MTR Head",
      "hsn_sac": "84137010",
      "gst_rate": "5 %",
      "quantity": 1,
      "unit": "Nos.",
      "rate_per_unit": 41303.02,
      "amount": 41303.02
    }
  ],
  "summary": {
    "total_taxable_value": 964438.43,
    "cgst_amount": 39521.30
  },
  "additional_info": {
    "jurisdiction": "JAIPUR",
    "invoice_type": "Computer Generated Invoice"
  }
}
```

----

### 20 image to takes approx 15 min

In [30]:
import base64
import google.generativeai as genai
from IPython.display import Markdown

# Configure Gemini (make sure you have your API key set up)
genai.configure(api_key="AIzaSyAMlYqwvuQgekl8nlqc56XTqJVBufszrBU")

def encode_image(image_path):
    """Encode image to base64"""
    with open(image_path, "rb") as image_file:
        return base64.b64encode(image_file.read()).decode('utf-8')

def process_single_image(image_path, system_prompt, user_prompt):
    """Process a single image with Gemini"""
    # Encode the image
    base64_image = encode_image(image_path)
    
    # Create the model
    model = genai.GenerativeModel('gemini-2.5-flash')
    
    # Prepare the content
    image_part = {
        "mime_type": "image/png",  # or "image/jpeg" depending on your image
        "data": base64_image
    }
    
    # Generate content
    response = model.generate_content([
        system_prompt,
        image_part,
        user_prompt
    ])
    
    return response.text

def process_multiple_images(image_paths, system_prompt, user_prompt):
    """Process multiple images and combine results"""
    all_results = []
    
    for image_path in image_paths:
        print(f"Processing: {image_path}")
        try:
            result = process_single_image(image_path, system_prompt, user_prompt)
            all_results.append({
                "image_path": image_path,
                "result": result
            })
        except Exception as e:
            print(f"Error processing {image_path}: {e}")
            all_results.append({
                "image_path": image_path,
                "error": str(e)
            })
    
    return all_results

# Your prompts
system_prompt = """You are a specialist in comprehending receipts.
Input images in the form of receipts will be provided to you,
and your task is to respond to questions based on the content of the input image."""

user_prompt = "Convert Invoice data into json format with appropriate json tags as required for the data in image"

image_paths = [
    "/home/manish/Desktop/projects/etfp/temp/image_19.png",
    "/home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png",
    "/home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg",
    "/home/manish/Desktop/projects/etfp/temp/image_20.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_0.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_1.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_2.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_3.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_4.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_5.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_6.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_7.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_8.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_9.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_10.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_11.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_12.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_13.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_14.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_15.png",
    "/home/manish/Desktop/projects/etfp/images/train/image_16.png",
    
]

# Process all images
results = process_multiple_images(image_paths, system_prompt, user_prompt)

# Display results
for result in results:
    print(f"\n=== Results for {result['image_path']} ===")
    if 'result' in result:
        display(Markdown(result['result']))
    else:
        print(f"Error: {result['error']}")

Processing: /home/manish/Desktop/projects/etfp/temp/image_19.png
Processing: /home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png
Processing: /home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg
Processing: /home/manish/Desktop/projects/etfp/temp/image_20.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_0.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_1.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_2.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_3.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_4.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_5.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_6.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_7.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_8.png
Processing: /home/manish/Desktop/projects/etfp/images/train/image_9.

```json
{
  "invoice_header": {
    "invoice_number": "12",
    "date": "6-May-2021",
    "page_number": "3",
    "supplier_reference": "12",
    "delivery_note": null,
    "mode_terms_of_payment": null,
    "other_references": null,
    "buyer_order_no": null,
    "despatch_document_no": null,
    "delivery_note_date": null,
    "despatched_through": null,
    "destination": "RAJGARH",
    "bill_of_lading_rr_no": null,
    "motor_vehicle_no": null,
    "terms_of_delivery": null
  },
  "biller_details": {
    "name": "High Furnace Manufacturer",
    "address": "Unit 1, Sector 2, Khanpur road Jaisalmer, Rajasthan.",
    "gst_number": "23AM2359012D1ZW"
  },
  "consignee_details": {
    "name": "JK Energies Limited",
    "address": "House No. 1, Kolar road, Rajgarh, Madhya Pradesh.",
    "gst_number": "23KM4589012B1ZS"
  },
  "line_items": [
    {
      "hsn_sac": "85414012",
      "taxable_value": 486641.28,
      "central_tax": {
        "rate": 2.50,
        "amount": 12166.02
      },
      "state_tax": {
        "rate": 2.50,
        "amount": 12166.02
      },
      "total_tax_amount_item": 24332.04
    },
    {
      "hsn_sac": "730890",
      "taxable_value": 162304.77,
      "central_tax": {
        "rate": 9.00,
        "amount": 14607.43
      },
      "state_tax": {
        "rate": 9.00,
        "amount": 14607.43
      },
      "total_tax_amount_item": 29214.86
    },
    {
      "hsn_sac": "841391",
      "taxable_value": 74777.69,
      "central_tax": {
        "rate": 9.00,
        "amount": 6729.99
      },
      "state_tax": {
        "rate": 9.00,
        "amount": 6729.99
      },
      "total_tax_amount_item": 13459.98
    },
    {
      "hsn_sac": "84137010",
      "taxable_value": 240714.69,
      "central_tax": {
        "rate": 2.50,
        "amount": 6017.86
      },
      "state_tax": {
        "rate": 2.50,
        "amount": 6017.86
      },
      "total_tax_amount_item": 12035.72
    }
  ],
  "summary_totals": {
    "total_taxable_value": 964438.43,
    "total_central_tax_amount": 39521.30,
    "total_state_tax_amount": 39521.30,
    "grand_total_tax_amount": 79042.60,
    "sgst_r_off_amount": 39521.30,
    "less_adjustment": -0.03,
    "amount_chargeable": 1043481.00,
    "amount_chargeable_in_words": "INR Ten Lakh Forty Three Thousand Four Hundred Eighty One Only",
    "tax_amount_in_words": "INR Seventy Nine Thousand Forty Two and Sixty paise Only"
  },
  "bank_details": {
    "bank_name": "IDFC FIRST BANK",
    "account_number": null,
    "ifsc_code": null
  },
  "other_information": {
    "company_pan": null,
    "jurisdiction": "JAIPUR",
    "is_computer_generated_invoice": true
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/temp/Bill Image_Receipt.png ===


```json
{
  "store_info": {
    "name": "The Lone Pine",
    "address": "43 Manchester Road\n12480 Brisbane\nAustralia",
    "phone_number": "617-3236-6207"
  },
  "transaction_details": {
    "invoice_number": "08000008",
    "date": "09/04/08",
    "time": "12:45",
    "table_number": "25"
  },
  "items": [
    {
      "quantity": 2,
      "description": "Carlsberg Bottle",
      "price": 16.00
    },
    {
      "quantity": 3,
      "description": "Heineken Draft Standard",
      "price": 24.60
    },
    {
      "quantity": 1,
      "description": "Heineken Draft Half Liter",
      "price": 15.20
    },
    {
      "quantity": 2,
      "description": "Carlsberg Bucket (5 bottles)",
      "price": 80.00
    },
    {
      "quantity": 4,
      "description": "Grilled Chicken Breast",
      "price": 74.00
    },
    {
      "quantity": 3,
      "description": "Sirloin Steak",
      "price": 96.00
    },
    {
      "quantity": 1,
      "description": "Coke",
      "price": 3.50
    },
    {
      "quantity": 5,
      "description": "Ice Cream",
      "price": 18.00
    }
  ],
  "summary": {
    "subtotal": 327.30,
    "sales_gov_tax": {
      "percentage": "5%",
      "amount": 16.36
    },
    "service_charge": {
      "percentage": "10%",
      "amount": 32.73
    },
    "grand_total": 376.40
  },
  "payment_info": {
    "cash_paid": 400.00,
    "change": 23.60
  },
  "footer": {
    "greeting": "Thank you and see you again!",
    "staff_name": "John",
    "promotional_message": "Bring this bill back within the next 10 days and get 15% discount on that day's food bill..."
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/temp/good-hand-written-bill.jpg ===


```json
{
  "business_name": "VISHAL PUNJABI VAISHNO BHOJANALYA",
  "business_description": "Real taste of Punjabi food, South Indian, North India, Jain Food, Veg, Chinese PURE VEGETARIAN FAMILY RESTAURANT",
  "address": "HOTEL SUN-N-SUN, BOULEVARD ROAD, G-2, SRINAGAR",
  "receipt_number": "1687",
  "date": "22/05/2007",
  "customer_info": {
    "to": "Cash",
    "recipient_name_handwritten": "Gopal"
  },
  "items": [
    {
      "quantity": 1,
      "description": "Khichri with Butter",
      "amount": 35.00
    },
    {
      "quantity": 1,
      "description": "Khoya Kaju",
      "amount": 50.00
    },
    {
      "quantity": 2,
      "description": "Curd",
      "amount": 24.00
    },
    {
      "quantity": 4,
      "description": "Roti",
      "amount": 12.00
    },
    {
      "quantity": 1,
      "description": "Rice",
      "amount": 25.00
    }
  ],
  "total_amount": 146.00,
  "total_amount_in_words": "Rs 146/- only",
  "currency": "INR",
  "notes": "Spl. Arrangement for Packing",
  "thank_you_message": "Thank you."
}
```


=== Results for /home/manish/Desktop/projects/etfp/temp/image_20.png ===


```json
{
  "invoice_header": {
    "invoice_no": "12",
    "dated": "6-May-2021",
    "delivery_note": null,
    "mode_terms_of_payment": null,
    "supplier_ref": "12",
    "other_reference": null,
    "buyer_order_no": null,
    "buyer_order_date": null,
    "despatch_document_no": null,
    "delivery_note_date": null,
    "despatched_through": null,
    "destination": "RAJGARH",
    "bill_of_lading_rr_no": null,
    "motor_vehicle_no": null,
    "terms_of_delivery": null,
    "page_info": "Page 2"
  },
  "biller_info": {
    "name": "High Furnace Manufacturer",
    "address": "Unit 1, Sector 2, Khanpur road Jaisalmer, Rajasthan.",
    "gst": "23AM2359012D1ZW"
  },
  "consignee_info": {
    "name": "JK Energies Limited",
    "address": "House No. 1, Kolar road, Rajgarh, Madhya Pradesh.",
    "gst": "23KM4589012B1ZS"
  },
  "line_items": [
    {
      "sl_no": 14,
      "description_of_goods": "DC Submersible Pumps 3 HP 30 MTR Head",
      "hsn_sac": "84137010",
      "gst_rate": "5 %",
      "quantity": "1 Nos.",
      "rate": "36934.54",
      "amount": "36934.54"
    },
    {
      "sl_no": 15,
      "description_of_goods": "SPV MODULES 24V 335W POLY",
      "hsn_sac": "85414012",
      "gst_rate": "5 %",
      "quantity": "9 Nos.",
      "rate": "6554.59",
      "amount": "58991.31"
    },
    {
      "sl_no": 16,
      "description_of_goods": "3 HP Capacity Solar Iron Structure",
      "hsn_sac": "730890",
      "gst_rate": "18 %",
      "quantity": "1 SET",
      "rate": "18353.39",
      "amount": "18353.39"
    },
    {
      "sl_no": 17,
      "description_of_goods": "BOS 3 HP PSMP @ 70 MTR",
      "hsn_sac": "841391",
      "gst_rate": "18 %",
      "quantity": "1 SET",
      "rate": "11834.51",
      "amount": "11834.51"
    },
    {
      "sl_no": 18,
      "description_of_goods": "DC Submersible Pumps 3 HP 70 MTR Head",
      "hsn_sac": "84137010",
      "gst_rate": "5 %",
      "quantity": "1 Nos.",
      "rate": "36376.56",
      "amount": "36376.56"
    },
    {
      "sl_no": 19,
      "description_of_goods": "SPV MODULES 24V 320W POLY",
      "hsn_sac": "85414012",
      "gst_rate": "5 %",
      "quantity": "15 Nos.",
      "rate": "6048.99",
      "amount": "90734.85"
    },
    {
      "sl_no": 20,
      "description_of_goods": "3 HP Capacity Solar Iron Structure",
      "hsn_sac": "730890",
      "gst_rate": "18 %",
      "quantity": "1 SET",
      "rate": "18178.98",
      "amount": "18178.98"
    },
    {
      "sl_no": 21,
      "description_of_goods": "2 HP Capacity Solar Iron Structure",
      "hsn_sac": "730890",
      "gst_rate": "18 %",
      "quantity": "1 SET",
      "rate": "12740.00",
      "amount": "12740.00"
    },
    {
      "sl_no": 22,
      "description_of_goods": "BOS 5 HP PSMP @ 100 MTR",
      "hsn_sac": "841391",
      "gst_rate": "18 %",
      "quantity": "1 SET",
      "rate": "18722.68",
      "amount": "18722.68"
    },
    {
      "sl_no": 23,
      "description_of_goods": "DC Submersible Pumps 5 HP 100 MTR Head",
      "hsn_sac": "84137010",
      "gst_rate": "5 %",
      "quantity": "1 Nos.",
      "rate": "41303.02",
      "amount": "41303.02"
    }
  ],
  "summary_info": {
    "subtotal": "964438.43",
    "cgst": "39521.30",
    "total_amount_net": null,
    "total_amount_gross": null
  },
  "other_notes": [
    "continued ...",
    "SUBJECT TO JAIPUR JURISDICTION",
    "This is a Computer Generated Invoice"
  ]
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_0.png ===


```json
{
  "document_type": "Tax Invoice",
  "document_status": "Original",
  "seller_details": {
    "legal_name": "Dell International Services India Pvt. Ltd.",
    "gst_reg_no": "29AAACH1925Q1Z6",
    "pan_no": "AAACH1925Q",
    "address": "12/1, 12/2A, 13/1A, Divya Sree Greens, Challaghata Village, Varthur Hobli, Domlur Post, Inner Ring Road, Bangalore 560 071",
    "ship_from_state": "KA"
  },
  "billing_address": {
    "company_name": "Hindalco Industries Limited",
    "contact_person": "Mr ABHISHEK SHARM",
    "phone": "09702052428",
    "address_line1": "2ND FLOOR, C-WING, PARAGON CENTRE,",
    "address_line2": "PANDURANG BUDHKAR MARG, WORLI,",
    "city": "MUMBAI",
    "pin_code": "400030",
    "state": "MH",
    "country": "India"
  },
  "shipping_address": {
    "company_name": "Hindalco Industries Limited",
    "contact_person": "Mr ABHISHEK SHARM",
    "phone": "09702052428",
    "address_line1": "2ND FLOOR, C-WING, PARAGON CENTRE,",
    "address_line2": "PANDURANG BUDHKAR MARG, WORLI,",
    "city": "MUMBAI",
    "pin_code": "400030",
    "state": "MH",
    "country": "India"
  },
  "invoice_details": {
    "invoice_no": "1410515599",
    "customer_no": "791114435",
    "order_no": "851893966",
    "page": "2 of 5",
    "bill_to_gst_reg_no": "27AAACH1201R1ZN",
    "ship_to_gst_reg_no": "27AAACH1201R1ZN",
    "bill_to_pan": "AAACH1201R",
    "ship_to_pan": "AAACH1201R",
    "customer_po": "SP-IT\\MUM-40\\20-21",
    "quotation_no": "30426645",
    "payment_terms": "ZINAB30 DaysINV",
    "due_date": "25/09/2020",
    "invoice_date": "27/07/2020",
    "delivery_type": "INTERNATIONAL CARRIER",
    "order_type": "I3 System order",
    "salesperson": "GOURAV KUMAR",
    "shipping_method": "BY-AIR"
  },
  "line_items": [
    {
      "description": "Dell Latitude 3301 CTO",
      "place_of_supply": "MAHARASHTRA (27)",
      "hsn_sac": "84713010",
      "quantity": 1,
      "unit_price": 51207.00,
      "net_amount": 51207.00,
      "tax_type": "IGST",
      "tax_rate": 18.00,
      "tax_amount": 9217.26,
      "features": [
        "Integrated Intel(R) UHD Graphics 620 for i5-8265U Processor with 8GB Memory",
        "Thank you for buying Dell",
        "8th Generation Intel Core i5-8265U Processor (4 Core,6MB Cache,1.6GHz up to 3.9GHz)",
        "ENERGY STAR Qualified",
        "13.3 HD (1366 x 768) Anti-Glare Non-Touch, Camera & Microphone, WLAN/WWAN Capable, Black",
        "Palmrest with no security, backlit",
        "Black LCD Cover",
        "Black Bottom Door, WLAN",
        "No AutoPilot",
        "Dell Applications for Windows 10",
        "Document for MUI (English, Khemer, Bahas)",
        "Quick Start Guide for 3301",
        "India BIS Package Label",
        "System Shipment, Latitude 3301 RIN P114G901 (India)",
        "ICC and India Countries Information",
        "8GB, onboard, LPDDR3, 2133MHz",
        "OS Media Kit Not Included",
        "SSD Bracket",
        "M.2 512GB PCIe NVMe Class 35 Solid State Drive",
        "65 Watt AC Adapter",
        "Power Cord for 3-pin Adapter (IND)",
        "4 Cell 52WHr Battery",
        "(INDIA) Dell Pro Backpack 15 ? PO1520P ? Fits most laptops up to 15\"",
        "Intel Core(TM) i5 Processor Label",
        "Not EPEAT Registered",
        "Wireless 9462 card",
        "Wireless Bluetooth Driver",
        "No Mobile Broadband Card",
        "Dark Grey Backlit Keyboard, English International",
        "Windows 10 Pro (64bit) English",
        "No Software",
        "Dell Developed Recovery Environment",
        "No Anti-Virus Software",
        "No Installation Service Required",
        "(CS-I) ProDeploy Deployment Dell Client PC System Configuration",
        "SI EMEA MOD INFO CFI ROUTING",
        "ProDeploy, Factory Install, Config Services",
        "Integration Information",
        "SI,MOD,INFO,SO,IMAGE,SI INJECT",
        "4Yr Keep Your Hard Drive",
        "ProDeploy Dell Client PC Deployment",
        "Prodeploy Deployment Dell Client PC",
        "Not selected in this configuration"
      ]
    }
  ]
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_1.png ===


```json
{
  "invoice_details": {
    "seller_info": {
      "name": "VIGIL Safety Technologies",
      "address": "PLOT NO.59,SECTOR No.1, 1st FLOOR SHARADA NIWAS, SUBHASH NAGAR, BELAGAVI - 590016",
      "contact_no": "0831-2451566",
      "email": "info@vigiltron.com",
      "gstin": "29BXLKPK9082K1ZL"
    },
    "buyer_info": {
      "name": "M/s HINDALCO INDUSTRIES LTD.",
      "address": "NEHRU NAGAR POST, Belgaum",
      "party_gstin": "29AAACH1201R2ZI"
    },
    "invoice_numbers": {
      "invoice_no": "VST/783",
      "inv_date": "12-07-2022",
      "po_no": "13542310381"
    },
    "line_items": [
      {
        "s_n": 1,
        "description_of_goods": "4U Rack 500 D\nModel: With Accessories\nSerial:",
        "hsn": null,
        "rate": 3040.00,
        "dsc_percent": "0+0+0",
        "price": 3040.00,
        "qty": "2 Nos",
        "net_amt": 6080.00,
        "gst_percent": "18%",
        "amount": 7174.40
      }
    ],
    "summary": {
      "sub_total": 7174.40,
      "rounded_off": 0.40,
      "tax_breakdown": [
        {
          "tax_rate": "18%",
          "taxable_amt": 6080.00,
          "cgst_amt": 547.20,
          "sgst_amt": 547.20,
          "total_tax": 1094.40
        }
      ],
      "grand_total": 7174.00,
      "grand_total_in_words": "Rupees Seven Thousand One Hundred Seventy Four Only"
    },
    "bank_details": {
      "bank_name": "STATE BANK OF INDIA",
      "branch": "SHIVBASAV NAGAR BRANCH",
      "account_no": "35801239891",
      "ifsc_code": "SBIN0008789"
    },
    "terms_and_conditions": [
      "1.Warranty 12 months from invoice date, Excluding physical damages and Burnt item due to natural climates and power surpluses Etc.",
      "2.Interest @24%p.a will be charged 30days of sale, if not paid within",
      "3.Goods Once Sold will not be Taken Back or Exchanged.",
      "4.Yearly 6 Services free.",
      "SUBJECT TO BELGAUM JURISDICTION"
    ],
    "declaration": "We declare that this invoice shows the actual price of the goods described and that all particulars are true and correct."
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_2.png ===


```json
{
  "invoice_details": {
    "invoice_number": "AMC/2122/FA/0019",
    "invoice_date": "28-09-2021",
    "bill_paid_date": "21 OCT 2021"
  },
  "seller_details": {
    "name": "Aditya Birla Sun Life AMC Ltd.",
    "gstin": "27AAACB6134D1Z4",
    "pan": "AAACB6134D",
    "address": "One World Center, Tower-1, 17th floor, Jupiter Mill Compound, 841, Senapati Bapat Marg, Elphinstone Road, Mumbai 400013",
    "state_code": "27",
    "phone_numbers": [
      "+91 22 4356 8000",
      "+91 22 4356 8110/8111"
    ],
    "website": "www.adityabirlasunlifeamc.com",
    "cin": "U65991MH1994PLC080811"
  },
  "buyer_details": {
    "name": "Hindalco Industries Limited",
    "address": "B Wing, 6, 7, Birla Centurion, Pandurang Budhkar Marg, Worli, Mumbai - 400030, Maharashtra",
    "state_code": "27",
    "gstin": "27AAACH1201R1ZN",
    "place_of_supply": "MAHARASHTRA"
  },
  "items": [
    {
      "sr_no": 1,
      "particulars": "Sale of Old Car of Lalit Kumar Apparao Kodi",
      "hsn": "87032291",
      "amount": 252495.00
    }
  ],
  "summary": {
    "taxable_value": 249352.00,
    "gst_on_margin_rs": 249352.00,
    "sgst": {
      "percentage": "6%",
      "amount": 14961.00
    },
    "cgst": {
      "percentage": "6%",
      "amount": 14961.00
    },
    "total_gst": 29922.00,
    "grand_total": 282417.00,
    "grand_total_in_words": "Two Lacs Eighty Two Thousand Four Hundred and Seventeen Only"
  },
  "tax_details": {
    "gst_rate_applied": "12%",
    "tax_payable_under_reverse_charge": "No"
  },
  "bank_details": {
    "bank_name": "HDFC Bank Ltd",
    "branch": "Nanik Motwani Marg, Fort, Mumbai - 400 001",
    "account_number": "00602090001619",
    "ifsc_code": "HDFC0000060"
  },
  "authorised_signatory": "Mohana"
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_3.png ===


```json
{
  "invoice_details": {
    "invoice_type": "Tax Invoice - E-Invoice Excel",
    "original_for": "RECIPIENT",
    "invoice_number": "HOA/SAVS/21/145",
    "invoice_date": "16/10/2021",
    "irn": "1599f01a7a42d25fab490db48c2da96a367994819369305369659beaf6555bb",
    "generated_on": "21/10/2021, 12:13 PM"
  },
  "seller_details": {
    "name": "KRISH CARS PVT. LTD.",
    "address": "G7-10, SHALIMAR MOURYA ESTATE OPP:INFINITY MALL, NEW LINK ROADANDHERI (W),MUMBAI -400053 MAHARASHTRA",
    "gstin": "27AADCk4616L1ZC",
    "state_code": "27",
    "pan_no": "AADCK4616L",
    "cin_no": "U34100NH2008PTC184513"
  },
  "customer_details": {
    "name": "HINDALCO INDUSTRIES LIMITED - kcpl",
    "billing_address": {
      "address_line_1": "7TH FLOOR, BIRLA CENTURION PANDURANG, BHUDKAR ROAD, WORLI, MUMBAI",
      "pincode": "400030",
      "state": "Maharashtra",
      "country": "INDIA"
    },
    "mobile": null,
    "state_code": "27",
    "pan": "AAACH1201R",
    "gst_no": "27AAACH1201R1ZN"
  },
  "place_of_supply": {
    "address": {
      "address_line_1": "7TH FLOOR, BIRLA CENTURION PANDURANG, BHUDKAR ROAD, WORLI, MUMBAI",
      "pincode": "400030",
      "state": "Maharashtra",
      "country": "INDIA"
    },
    "mobile": null,
    "state_code": "27",
    "state": "Maharashtra"
  },
  "vehicle_details": {
    "vehicle_description": "CITY 1.5 5TH GEN VX CVT (i-VTEC)-PLATINUM WHITE P",
    "color": "PLATINUM WHITE P.",
    "vin_no": "MAKGN264GM4001645",
    "engine_no": "L15ZD1223209",
    "chasis_no": "MAKGN264GM4001645",
    "financed_by": "N.A"
  },
  "line_items": [
    {
      "sn": 1,
      "item_name": "VEHICLE SALES",
      "hsn_sac_code": "87032291",
      "qty": 1.00,
      "rate": 971280.00,
      "total_item_value": 971280.00,
      "discount": 41379.00,
      "taxable_amount": 929901.00,
      "tax_details": {
        "sgst_percentage": 14.00,
        "sgst_amount": 130186.14,
        "cgst_percentage": 14.00,
        "cgst_amount": 130186.14,
        "igst_percentage": 0.00,
        "igst_amount": 0.00,
        "cess_percentage": 17.00,
        "cess_amount": 158083.17,
        "total_tax_amount": 418455.45
      }
    }
  ],
  "summary": {
    "no_of_line_items": 1,
    "total_qty": 1.00,
    "sub_total": 971280.00,
    "less_discount": 41379.00,
    "total_taxable_amount": 929901.00,
    "total_cgst": 130186.14,
    "total_sgst": 130186.14,
    "total_igst": 0.00,
    "total_cess": 158083.17,
    "ex_showroom_amount": 1348356.00,
    "tcs_amount": 13483.56,
    "invoice_amount": 1361840.00,
    "total_invoice_value_in_words": "Rupees Thirteen Lakh Sixty One Thousand Eight Hundred Fourty Only"
  },
  "remark": "Tax Invoice for car procured for M/s. Abhishek Rungla @hawan",
  "signature": {
    "customer_signature": null,
    "authorized_signatory": "KRISH CARS PVT. LTD. (KORIVALI)"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_4.png ===


```json
{
  "invoice_details": {
    "invoice_number": "227/2019-2020",
    "invoice_date": "5th June 2019",
    "reverse_charge_applicable": "N",
    "place_of_supply": "Bhubaneshwar",
    "state_of_supply": "Odisha",
    "state_code_of_supply": "21",
    "type": "Original for Recipient"
  },
  "seller_details": {
    "name": "Space Forte",
    "address": "Office No.8,16,17, 38-West Building, Road No.6, Sion (W), Mumbai 400 022",
    "gstin": "27AACPK9036G1ZP",
    "pan_no": "AACPK9036G",
    "tel": "24084924 / 24037181",
    "registered_office_details": [
      {
        "location": "Mumbai",
        "address": "No. 8/16/17, 38 - West, Road No. 6, Sion (W), Mumbai - 400 022",
        "tel": "+91 22 24037181, 24084924",
        "fax": "+91 22 24072043"
      },
      {
        "location": "Pune",
        "address": "E/9 Hermes Park, 30 Bund Garden Road, Camp, Pune - 411 011",
        "tel": "+91 20 26161987",
        "email": "skkassociates@gmail.com"
      }
    ]
  },
  "bill_to_party": {
    "name": "M/s Hindalco Industries Ltd",
    "address": "6th & 7th Floor, Birla Centurion, Pandurang Budhkar Marg, Worli, Mumbai 400 030",
    "gstin": "27AAACH1201R1ZN",
    "pan_no": "AAACH1201R",
    "state": "Maharashtra",
    "state_code": "27"
  },
  "ship_to_party": {
    "name": "M/s Hindalco Industries Ltd",
    "address": "Gandhi Peace Centre, Khandagiri, Bhubaneshwar, Odisha",
    "gstin": null,
    "pan_no": null,
    "state": "Odisha",
    "state_code": "21"
  },
  "line_items": [
    {
      "s_no": null,
      "product_description": "Reference PO No: HIL/MMM-04/2018-19",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": null,
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "A",
      "product_description": "Civil work",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "19,25,475.33",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "B",
      "product_description": "Carpentry work",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "12,29,859.75",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "C",
      "product_description": "Electrical work",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "3,20,275.00",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "D",
      "product_description": "POP and Gypsum ceiling",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "5,79,227.38",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "F",
      "product_description": "Painting & Melamine Polish",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "2,49,469.00",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "G",
      "product_description": "Glass Roofing System",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "30,00,000.00",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "H",
      "product_description": "HVAC",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "2,21,000.00",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": "I",
      "product_description": "Miscellaneous Items",
      "hsn_code": null,
      "qty": null,
      "rate": null,
      "amount": "1,47,500.00",
      "discount": null,
      "taxable_value": null
    },
    {
      "s_no": null,
      "product_description": "Interior works contract services as per items enlisted above",
      "hsn_code": "995479",
      "qty": null,
      "rate": null,
      "amount": "76,72,806.46",
      "discount": null,
      "taxable_value": "76,72,806.46"
    }
  ],
  "totals": {
    "total_amount_before_tax": "76,72,806.46",
    "igst": {
      "rate": "18.00%",
      "amount": "13,81,105.16"
    },
    "total_amount_after_tax": "90,53,911.62",
    "round_off": "90,53,912.00",
    "gst_on_reverse_charge": "0.00",
    "total_invoice_amount_in_words": "Rupees Ninety Lakhs Fifty Three Thousand Nine Hundred Twelve Only"
  },
  "bank_details": {
    "bank_name": "Kotak Mahindra Bank",
    "bank_account_number": "2711914454",
    "bank_ifsc": "KKBK0000635"
  },
  "terms_and_conditions": [
    "18% interest will be charged on all invoices not paid within 30 days from date of invoice."
  ],
  "authorized_signatory": {
    "name": "Sundeep Kankani"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_5.png ===


```json
{
  "invoice_header": {
    "invoice_type": "e-Invoice / Tax Invoice",
    "irn_no": "8919f9c0ac334bbdc7e8197e642ec8685587df39005a71d1c019796b64df4c12",
    "copy_type": [
      "Original for Recipient",
      "Duplicate for Transporter",
      "Triplicate for Supplier"
    ]
  },
  "seller_details": {
    "dispatched_from": "LAUREN INFORMATION TECHNOLOGIES PVT. LTD.",
    "address": "14th Khar Danda Road Jains Arcade, 3rd Floor Khar West MUMBAI-400052 Maharashtra India",
    "gstin": "27AAACL5292L1Z4",
    "gst_type": "Regular/TDS/ISD",
    "pan_no": "AAACL5292L",
    "cin": "U72200MH1992PTC069851",
    "lut": "AD2703230606470"
  },
  "buyer_details": {
    "name": "Hindalco Industries Ltd.",
    "address": "Post Office:- Post Bag No 1, Yamanapur Village - Belagavi Works, Hindalco Industries Limited Belagavi - 590010 KA",
    "gstin": "29AAACH1201R2ZI",
    "gst_type": "Regular/TDS/ISD",
    "place_of_supply": "Karnataka_KA",
    "state_code": "29",
    "pan_no": "AAACH1201R"
  },
  "delivery_details": {
    "name": "Hindalco Industries Ltd.",
    "address": "Post Office:- Post Bag No 1, Yamanapur Village - Belagavi Works, Hindalco Industries Limited Belagavi - 590010 KA",
    "gstin": "29AAACH1201R2ZI",
    "gst_type": "Regular/TDS/ISD",
    "place_of_supply": "Karnataka_KA",
    "state_code": "29",
    "pan_no": "AAACH1201R"
  },
  "invoice_details": {
    "invoice_no": "MHAR2425 - 390036",
    "invoice_date": "10-04-2024",
    "place_of_supply": "Maharashtra",
    "supply_date": "10-04-2024",
    "cust_order_no": "13542319324",
    "place_of_supply_cust": ""
  },
  "bank_details": {
    "bank_name": "ICICI Bank Limited",
    "account_no": "121751000007",
    "account_type": "Current Account",
    "address": "ICICI Bank Ltd, E Tower, Ground floor, Bharat Diamond Bourse, Bandra Kurla Complex, Bandra East, Mumbai - 400 051",
    "ifsc_code": "ICIC0001217",
    "swift_code": "ICICINBBCTS",
    "bsr_code": null,
    "micr_code": "400229203",
    "bank_code": "0007"
  },
  "line_items": [
    {
      "sr_no": 1,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998713",
      "quantity": 1,
      "uom": null,
      "rate": 360000.00,
      "total_amount": 360000.00,
      "discount_amount": 0.00,
      "taxable_amount": 360000.00,
      "cgst": {
        "rate": 0.0,
        "amount": 0.00
      },
      "sgst_utgst": {
        "rate": 0.00,
        "amount": 0.00
      },
      "igst": {
        "rate": 18.0,
        "amount": 64800.00
      },
      "tcs": {
        "rate": 0.000,
        "amount": 0.00
      },
      "total_inr": 424800.00
    },
    {
      "sr_no": 2,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998713",
      "quantity": 1,
      "uom": null,
      "rate": 0.00,
      "total_amount": 0.00,
      "discount_amount": 0.00,
      "taxable_amount": 0.00,
      "cgst": {
        "rate": 0.0,
        "amount": 0.00
      },
      "sgst_utgst": {
        "rate": 0.00,
        "amount": 0.00
      },
      "igst": {
        "rate": 18.0,
        "amount": 0.00
      },
      "tcs": {
        "rate": 0.000,
        "amount": 0.00
      },
      "total_inr": 0.00
    },
    {
      "sr_no": 3,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998713",
      "quantity": 1,
      "uom": null,
      "rate": 0.00,
      "total_amount": 0.00,
      "discount_amount": 0.00,
      "taxable_amount": 0.00,
      "cgst": {
        "rate": 0.0,
        "amount": 0.00
      },
      "sgst_utgst": {
        "rate": 0.00,
        "amount": 0.00
      },
      "igst": {
        "rate": 18.0,
        "amount": 0.00
      },
      "tcs": {
        "rate": 0.000,
        "amount": 0.00
      },
      "total_inr": 0.00
    },
    {
      "sr_no": 4,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998713",
      "quantity": 1,
      "uom": null,
      "rate": 0.00,
      "total_amount": 0.00,
      "discount_amount": 0.00,
      "taxable_amount": 0.00,
      "cgst": {
        "rate": 0.0,
        "amount": 0.00
      },
      "sgst_utgst": {
        "rate": 0.00,
        "amount": 0.00
      },
      "igst": {
        "rate": 18.0,
        "amount": 0.00
      },
      "tcs": {
        "rate": 0.000,
        "amount": 0.00
      },
      "total_inr": 0.00
    }
  ],
  "footer": {
    "document_origin": "This is SAP generated document.",
    "jurisdiction": "Subject to MUMBAI Jurisdiction"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_6.png ===


```json
{
  "document_type": "DELIVERY CHALLAN",
  "document_copies": {
    "original_for_receipient": true,
    "duplicate_for_transporter": false,
    "triplicate_for_supplier": false
  },
  "dispatched_from": {
    "name": "LAUREN INFORMATION TECHNOLOGIES PVT. LTD.",
    "address": "14th Khar Danda Road Jains Arcade, 3rd Floor Khar West MUMBAI-400052 Maharashtra India",
    "gstin": "27AAACL5292L1Z4",
    "gst_type": "Regular/TDS/ISD",
    "pan_no": "AAACL5292L",
    "cin": "U72200MH1992PTC069851"
  },
  "buyer": {
    "name": "Hindalco Industries Ltd.",
    "address": "Post Office:- Post Bag No 1, Yamanapur Village - Belagavi Works, Hindalco Industries Limited Belagavi - 590010 KA"
  },
  "delivery_to": {
    "name": "Hindalco Industries Ltd.",
    "address": "Post Office:- Post Bag No 1, Yamanapur Village - Belagavi Works, Hindalco Industries Limited Belagavi - 590010 KA",
    "gstin": "29AAACH1201R2ZI",
    "gst_type": "Regular/TDS/ISD",
    "place_of_supply": "Karnataka_KA",
    "state_code": "29",
    "pan_no": "AAACH1201R"
  },
  "delivery_details": {
    "delivery_no": "MHDN2425 - 390019",
    "delivery_date": "10-04-2024",
    "place_of_supply_delivery": "Maharashtra",
    "customer_order_no": "13542319324",
    "customer_order_date": "10-04-2024"
  },
  "items": [
    {
      "sr_no": 1,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998313",
      "uom": null,
      "quantity": 1
    },
    {
      "sr_no": 2,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998313",
      "uom": null,
      "quantity": 1
    },
    {
      "sr_no": 3,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998313",
      "uom": null,
      "quantity": 1
    },
    {
      "sr_no": 4,
      "item_code": "10002",
      "description_of_goods": "AMC for IBM Servers and tape library as per annexure attached",
      "hsn_sac_code": "998313",
      "uom": null,
      "quantity": 1
    }
  ],
  "total": {
    "quantity": 4
  },
  "terms_and_conditions": "Acceptance of Goods and Services under this invoice or making part or full payment or only acknowledgment of this invoice is treated as acceptance of Terms & Conditions printed overleaf and acceptance of Sales Terms & Conditions of Lauren Information Technologies Private Limited.",
  "declaration": "I/We hereby certify that my/our registration certificate under the GST Act, is in force on the date on which the sale of the goods specified in this tax invoice is made by me/us and that the transaction of sales covered by this Tax Invoice has been collected by me/us and it shall be accounted for in the turnover of sales while filing of return and the due tax, If any, payable on the sale has been paid or shall be paid",
  "certifying_company": "LAUREN INFORMATION TECHNOLOGIES PVT. LTD.",
  "authorized_signatory": {
    "name": "Charuchandra Waghmare",
    "title": "Authorised Signatory",
    "digital_signature_details": "Digitally signed by Charuchandra Waghmare DN: cn=Charuchandra Waghmare, o=Lauren Information Technologies Pvt. Ltd. Date: 2024.04.19 13:15:35 Z"
  },
  "document_generation_info": {
    "system": "SAP generated document",
    "jurisdiction": "MUMBAI"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_7.png ===


```json
{
  "invoice_details": {
    "payment_terms_days": "30",
    "amount_in_words": "Indian Rupees Four Lakhs Twenty-Four Thousand Eight Hundred Only"
  },
  "items": [
    {
      "sr_no": null,
      "item_code": null,
      "description_of_goods": null,
      "hsn_sac_code": null,
      "quantity": null,
      "uom": null,
      "rate": null,
      "total_amount_item": 360000.00,
      "discount_amount": 0.00,
      "taxable_amount_item": 360000.00,
      "cgst": {
        "rate": 0.00,
        "amount": 0.00
      },
      "sgst_utgst": {
        "rate": 0.00,
        "amount": 0.00
      },
      "igst": {
        "rate": null,
        "amount": 64800.00
      },
      "tcs": {
        "rate": null,
        "amount": 0.00
      },
      "total_inr_item": 424800.00
    }
  ],
  "totals": {
    "sub_total": 360000.00,
    "cgst_total": 0.00,
    "sgst_utgst_total": 0.00,
    "igst_total": 64800.00,
    "tcs_total": 0.00,
    "round_off": 0.00,
    "invoice_amount_inr": 424800.00
  },
  "important_notes": [
    "1. All cheques/Drafts will be payable in the name of Lauren Information Technologies Pvt Ltd within the due date failing which interest @ 18% per annum will be charged on the due amount from the due date till the date of actual receipt of payment."
  ],
  "remarks": "Price is inclusive of tax/ Period:1-4-24 to 31-3-25/ Payment terms: 7 days. Based On Deliveries 390019.",
  "terms_conditions": "Acceptance of Goods and Services under this invoice or making part or full payment or only acknowledgment of this invoice is treated as acceptance of Terms & Conditions printed overleaf and acceptance of Sales Terms & Conditions of Lauren Information Technologies Private Limited.",
  "declaration": "I/We hereby certify that my/our registration certificate under the GST Act, is in force on the date on which the sale of the goods specified in this tax invoice is made by me/us and that the transaction of sales covered by this Tax Invoice has been collected by me/us and it shall be accounted for in the turnover of sales while filing of return and the due tax, If any, payable on the sale has been paid or shall be paid",
  "seller_info": {
    "company_name": "LAUREN INFORMATION TECHNOLOGIES PVT. LTD.",
    "certified_by": "Charuchandra Waghmare",
    "authorised_signatory": "Charuchandra Waghmare",
    "digital_signature_details": {
      "signer": "Charuchandra Waghmare",
      "dn": "cn=Charuchandra Waghmare, o=Lauren Information Technologies Pvt. Ltd., c=US",
      "date": "2026.04.10",
      "time": "13:15:21 Z"
    }
  },
  "document_info": {
    "generation_source": "SAP generated document",
    "jurisdiction": "MUMBAI"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_8.png ===


```json
{
  "invoice_type": "Tax Invoice",
  "document_status": "Duplicate",
  "seller_information": {
    "name": "Dell International Services India Pvt. Ltd.",
    "registered_office_address": "12/1, 12/2A, 13/1A, Divyaa Sree Greens, Challaghata Village, Varthur Hobli, Domlur Post, Inner Ring Road, Bangalore 560 071",
    "gst_reg_no": "29AAACH1925Q1Z6",
    "pan_no": "AAACH1925Q",
    "ship_from_state": "KA",
    "telephone": "1800-425-4026 (Toll Free)",
    "website": "www.dell.co.in"
  },
  "billing_address": {
    "company_name": "Hindalco Industries Limited",
    "contact_person": "Mr ABHISHEK SHARM",
    "phone": "09702052428",
    "address_line1": "2ND FLOOR, C-WING, PARAGON CENTRE,",
    "address_line2": "PANDURANG BUDHKAR MARG, WORLI,",
    "city": "MUMBAI",
    "pincode": "400030",
    "state": "MH",
    "country": "India"
  },
  "shipping_address": {
    "company_name": "Hindalco Industries Limited",
    "contact_person": "Mr ABHISHEK SHARM",
    "phone": "09702052428",
    "address_line1": "2ND FLOOR, C-WING, PARAGON CENTRE,",
    "address_line2": "PANDURANG BUDHKAR MARG, WORLI,",
    "city": "MUMBAI",
    "pincode": "400030",
    "state": "MH",
    "country": "India"
  },
  "invoice_details": {
    "invoice_no": "1410515599",
    "customer_no": "791114435",
    "order_no": "851893966",
    "page_no": "1 of 5",
    "invoice_date": "27/07/2020",
    "delivery_type": "INTERNATIONAL CARRIER",
    "order_type": "I3 System order",
    "salesperson": "GOURAV KUMAR",
    "shipping_method": "BY- AIR",
    "bill_to_gst_reg_no": "27AAACH1201R1ZN",
    "ship_to_gst_reg_no": "27AAACH1201R1ZN",
    "bill_to_pan": "AAACH1201R",
    "ship_to_pan": "AAACH1201R",
    "cust_po": "SP-IT\\MUM-40\\20-21",
    "quotation_no": "30426645",
    "payment_terms": "ZINAB30 DaysINV",
    "due_date": "25/09/2020"
  },
  "note": "Thanks for placing order on Dell. Your Order is invoiced. Please find attached digitally signed Invoice."
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_9.png ===


```json
{
  "invoice_header": {
    "company_name": "DELL EMC",
    "gst_reg_no": "29AAACH1925Q1Z6",
    "pan_no": "AAACH1925Q",
    "trade_legal_name": "Dell International Services India Pvt. Ltd.",
    "seller_address": {
      "line1": "12/1, 12/2A, 13/1A, Divya Sree Greens,",
      "line2": "Challaghata Village, Varthur Hobli, Domlur Post,",
      "city": "Bangalore",
      "pincode": "560 071",
      "state": "Karnataka",
      "state_code": "KA"
    },
    "invoice_type": "Tax Invoice",
    "copy_type": "Original"
  },
  "billing_address": {
    "company_name": "Hindalco Industries Limited",
    "contact_person": "Mr ABHISHEK SHARM",
    "contact_number": "09702052428",
    "address_lines": [
      "2ND FLOOR, C-WING,",
      "PARAGON CENTRE,",
      "PANDURANG BUDHKAR MARG,"
    ],
    "locality": "WORLI",
    "city": "MUMBAI",
    "pincode": "400030",
    "state": "Maharashtra",
    "state_code": "MH",
    "country": "India"
  },
  "shipping_address": {
    "company_name": "Hindalco Industries Limited",
    "contact_person": "Mr ABHISHEK SHARM",
    "contact_number": "09702052428",
    "address_lines": [
      "2ND FLOOR, C-WING,",
      "PARAGON CENTRE,",
      "PANDURANG BUDHKAR MARG,"
    ],
    "locality": "WORLI",
    "city": "MUMBAI",
    "pincode": "400030",
    "state": "Maharashtra",
    "state_code": "MH",
    "country": "India"
  },
  "invoice_details": {
    "invoice_no": "1410515599",
    "customer_no": "791114435",
    "order_no": "851893966",
    "page": "3 of 5",
    "invoice_date": "27/07/2020",
    "delivery_type": "INTERNATIONAL CARRIER",
    "order_type": "I3 System order",
    "salesperson": "GOURAV KUMAR",
    "shipping_method": "BY- AIR",
    "bill_to_gst_reg_no": "27AAACH1201R1ZN",
    "ship_to_gst_reg_no": "27AAACH1201R1ZN",
    "bill_to_pan": "AAACH1201R",
    "ship_to_pan": "AAACH1201R",
    "cust_po": "SP-IT!MUM-40\\20-21",
    "quotation_no": "30426645",
    "payment_terms": "ZINAB30 DaysINV",
    "due_date": "25/09/2020"
  },
  "line_items": [
    {
      "description": "Custom Configuration",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "1Y Carry-In Service",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "1Y Basic Onsite Service Upgrade",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "Y2-4 Basic Onsite Service Extension",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "Visit //www.dell.com/contactdell",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "Dell Carries 1 Year Warranty from Invoice Date",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "Allows you to retain a defective hard drive that is still under warranty",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "NBD Onsite Labor Service only support limited OCSR parts replacement",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "For more detail, refer to Customer Self Repair webpage in Dell support website",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    },
    {
      "description": "FGA NB1-OU-BTO Freight Charges(India)",
      "place_of_supply": null,
      "hsn_sac": null,
      "quantity": 1,
      "unit_price": null,
      "net": null,
      "tax_type": null,
      "rate": null,
      "amount": null
    }
  ],
  "totals": {
    "currency": "INR",
    "subtotal": 51207.00,
    "tax_details": [
      {
        "tax_name": "IGST",
        "rate": "18%",
        "amount": 9217.26
      }
    ],
    "total_tax_amount": 9217.26,
    "total_value": 60424.26,
    "amount_in_words": "Rupees Sixty Thousand Four Hundred Twenty Four and Twenty Six Paise Only"
  },
  "other_information": {
    "tag_nos": "5HK0103"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_10.png ===


```json
{
  "invoice_type": "Tax Invoice",
  "document_nature": "Original for Receiver",
  "supplier_details": {
    "gstin_of_supplier": "24AACCG2599K1ZB",
    "cin": "U63013DL2004PLC217305",
    "name_of_supplier": "Go Airlines (India) Limited",
    "address_of_supplier": "SARADR PATEL NEW DOMESTIC BLDG., SARDAR NAGAR, AHMDABAD - 382475",
    "pan": "AACCG2599K",
    "pnr_no": "141HRV",
    "class": "Economy Class",
    "invoice_no": "2421542444141HRV",
    "invoice_date": "31/05/2019",
    "sector": "AMD-BOM",
    "place_of_supply": "27/Maharashtra"
  },
  "receiver_details": {
    "name": "Kotak Mahindra Bank Limited",
    "gstin_uin": "27AAAACK4409J1ZK",
    "address": "Maharashtra",
    "state_code": "27",
    "email_id_of_gstin_holder": "kotak.gstinput@gmail.com"
  },
  "line_items": [
    {
      "charge_code": "FarePrice",
      "charge_description": "* Fare",
      "harmonized_system_nomenclature_hsn": "9964",
      "description_of_service": "Domestic/International Scheduled Air transport services of passengers",
      "amount": 1774.00,
      "discount": 0.00,
      "net_amount": 1774.00,
      "igst_5_percent": 89.00,
      "total": 1863.00
    },
    {
      "charge_code": "PSF",
      "charge_description": "Passenger Service Fee",
      "harmonized_system_nomenclature_hsn": "--",
      "description_of_service": "--",
      "amount": 153.00,
      "discount": 0.00,
      "net_amount": 153.00,
      "igst_5_percent": null,
      "total": 153.00
    },
    {
      "charge_code": "UDF",
      "charge_description": "User Development Fee",
      "harmonized_system_nomenclature_hsn": "--",
      "description_of_service": "--",
      "amount": 100.00,
      "discount": 0.00,
      "net_amount": 100.00,
      "igst_5_percent": null,
      "total": 100.00
    }
  ],
  "summary": {
    "total_amount": 2027.00,
    "total_discount": 0.00,
    "total_net_amount": 2027.00,
    "total_igst_5_percent": 89.00,
    "grand_total": 2116.00
  },
  "notes": [
    "If you have opted for meal and insurance, you would get an invoice for the same directly from the service provider. Go Airlines has not supplied any goods/services against the collection of meal and insurance and just collected funds from you on behalf of the service provider.",
    "PSF/UDF/ADF or any airport charges/taxes by whatever name called are collected by GoAirlines on behalf of Airport Operator as a pure agent. Go Airlines has not supplied any service to passengers against collection of PSF/UDF/ADF or any other airport charges/taxes from passengers.",
    "(* Fare includes Base Fare, Fare Surcharge, Fuel (YQ), Passenger Handling Fees, Convenience Fees and Regional Connectivity Scheme.)"
  ],
  "authorized_signatory": "Go Airlines (India) Limited"
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_11.png ===


```json
{
  "header": {
    "company_name": "Hindalco Industries Limited",
    "main_office_address": "Flat/Door:- Belagavi Works, Hindalco Industries Limited, Post Office:- Post Bag No 1, Yamanapur Village, City/District:- Belagavi, State:- KARNATAKA, PIN Code:- 590010, Country:- India",
    "main_office_phone": "0831-2999145",
    "registered_office_address": "Ahura Centre, 1st Floor, B Wing, Mahakali Caves Road, Andheri(East), Mumbai-400093",
    "registered_office_email": "hindalco@adityabirla.com",
    "registered_office_phone_reg": "(91-22)6691 7000",
    "registered_office_phone_local": "07805 - 281007",
    "website": "www.hindalco.com",
    "document_type": "PURCHASE ORDER",
    "document_scope": "Domestic",
    "page_number": "1 of 4"
  },
  "purchase_order_details": {
    "po_number": "13542310381",
    "creation_date": "04-JUL-2022",
    "amendment_no": null,
    "amendment_date": null,
    "currency": "INR",
    "our_contact": "Sharanabasava Hiremath",
    "our_contact_email": "sharanabasava.h@adityabirla.com",
    "our_contact_phone": "+91 7353316276",
    "ship_to_location": "Belgavi Stores"
  },
  "vendor_details": {
    "vendor_code": "10028068",
    "vendor_name": "VIGIL SAFETY TECHNOLOGIES",
    "site": "S1-BELAGAVI",
    "address": "PLOT NO 59,SHARDA NIVAS,OPP MINORIT\nSUBHASH NAGAR MAIN ROAD\nBELAGAVI,\nBELAGAVI, KARNATAKA, India - 590010",
    "gstin_no": "29BXLK9082K1ZL",
    "email": null,
    "contact_person": null,
    "mobile": null
  },
  "supplier_quotation_reference": "IT REQ., QTN REF-313 DTD-16.06.22",
  "items": [
    {
      "s_no": 1,
      "item_code": "311623000035",
      "description": "MOUNT; TYPE: RACK, MFR P/N: WM-4-550400-WLD-Z6 MFR: NETRACK, MATERIAL: STEEL, WIDTH: 550 MM, LENGTH: 400 MM, MOUNTING METHOD: WALL, HEIGHT: 258 MM, STANDARD: DIN 41494, UL, USEFUL HEIGHT: 4U, WITH ALL ACCESSORIES",
      "uom": "NUMBER",
      "ship_qty": 2.000,
      "unit_rate": 3040.00,
      "amount": 6080.00
    }
  ],
  "summary": {
    "shipment_date_1": "26-JUL-2022",
    "basic_value": 6080.00,
    "item_total_value": 6080.00,
    "total_purchase_order_value": 6080.00,
    "value_in_words": "INR (Six Thousand Eighty Only)"
  },
  "other_terms_and_conditions": {
    "payment_terms": "Within 30 days from the date of receipt of material at site.",
    "price_type_terms": "Fixed",
    "freight_terms": "Freight Paid",
    "transportation_arrangement": null,
    "general_terms_and_conditions_note": "Our General Terms and Conditions are mentioned overleaf."
  },
  "company_info_footer": {
    "division": "Karnataka",
    "pan": "AAACH1201R",
    "cin_no": "L27020MH1958PLC011238",
    "import_export_code": "0388147237",
    "tan_no": "BLRH02471A",
    "gstin": "29AAACH1201R2ZI"
  },
  "authorized_signatory": {
    "name": "Narasimha Murthy A",
    "title": "Manager Purchase, Belagavi Works"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_12.png ===


```json
{
  "invoice_header": {
    "title": "Tax Invoice",
    "page": "3"
  },
  "biller_details": {
    "name": "M/S Avadhoot Paper Products Private Limited",
    "address": "Unit 1, Sector 2, Khanpur road Jaisalmer, Rajasthan.",
    "gstin": "23AM2359012D1ZW",
    "uam": "UDYAM-MH-26-0006537",
    "pan": "AAQCA0834K"
  },
  "consignee_details": {
    "name": "JK Energies Limited",
    "address": "House No. 1, Kolar road, Rajgarh, Madhya Pradesh.",
    "gstin": "23KM4589012B1ZS"
  },
  "invoice_details": {
    "invoice_no": "12",
    "date": "6-May-2021",
    "delivery_note": null,
    "mode_terms_of_payment": null,
    "supplier_ref": "12",
    "other_references": null,
    "buyer_order_no": null,
    "buyer_order_date": null,
    "despatch_document_no": null,
    "delivery_note_date": null,
    "despatched_through": null,
    "destination": "RAJGARH",
    "bill_of_lading_lr_rr_no": null,
    "motor_vehicle_no": null,
    "terms_of_delivery": null
  },
  "itemized_goods": [
    {
      "hsn_sac": "85414012",
      "taxable_value": 486641.28,
      "central_tax": {
        "rate": "2.50%",
        "amount": 12166.02
      },
      "state_tax": {
        "rate": "2.50%",
        "amount": 12166.02
      },
      "total_tax_amount": 24332.04
    },
    {
      "hsn_sac": "730890",
      "taxable_value": 162304.77,
      "central_tax": {
        "rate": "9%",
        "amount": 14607.43
      },
      "state_tax": {
        "rate": "9%",
        "amount": 14607.43
      },
      "total_tax_amount": 29214.86
    },
    {
      "hsn_sac": "841391",
      "taxable_value": 74777.69,
      "central_tax": {
        "rate": "9%",
        "amount": 6729.99
      },
      "state_tax": {
        "rate": "9%",
        "amount": 6729.99
      },
      "total_tax_amount": 13459.98
    },
    {
      "hsn_sac": "84137010",
      "taxable_value": 240714.69,
      "central_tax": {
        "rate": "2.50%",
        "amount": 6017.86
      },
      "state_tax": {
        "rate": "2.50%",
        "amount": 6017.86
      },
      "total_tax_amount": 12035.72
    }
  ],
  "totals": {
    "total_taxable_value": 964438.43,
    "total_central_tax_amount": 39521.30,
    "total_state_tax_amount": 39521.30,
    "grand_total_tax_amount": 79042.60,
    "sgst_cgst_summary_amount": 39521.30,
    "rounding_off": -0.03,
    "amount_chargeable": 1043481.00,
    "amount_chargeable_in_words": "INR Ten Lakh Forty Three Thousand Four Hundred Eighty One Only",
    "tax_amount_in_words": "INR Seventy Nine Thousand Forty Two and Sixty paise Only"
  },
  "bank_details": {
    "bank_name": "IDFC FIRST BANK",
    "account_no": null,
    "branch_ifs_code": null
  },
  "declaration": "We declare that this invoice shows the actual price of the goods described and that all particulars are true and correct.",
  "jurisdiction": "Jaipur",
  "additional_notes": "This is a Computer Generated Invoice"
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_13.png ===


```json
{
  "invoice_type": "TAX INVOICE",
  "document_originality": "ORIGINAL",
  "vendor_details": {
    "name": "ADITYA BIRLA MANAGEMENT CORPORATION PRIVATE LIMITED",
    "address": "C-1, Aditya Birla Centre, S K Ahire Marg, Worli, Mumbai 400030",
    "phone": "+91-22-24995000",
    "fax": "+91-22-24995740",
    "cin": "U73100MH1999PTC118379",
    "gstin": "27AAACB9318D1ZW",
    "pan": "AAACB9318D"
  },
  "bill_to_details": {
    "name": "HINDALCO INDUSTRIES LTD.",
    "address": "B-Wing, 1st floor, Ahura Centre, Mahakali Caves Road, Andheri(E), MUMBAI, MAHARASHTRA-400093",
    "gstin": "27AAACH1201R1ZN",
    "pan": "AAACH1201R",
    "place_of_supply": {
      "state": "MAHARASHTRA",
      "state_code": "27"
    }
  },
  "invoice_details": {
    "invoice_no": "MH0110200309",
    "invoice_date": "22.09.2020",
    "our_reference": "2100000035",
    "cust_id": "HIL0031"
  },
  "line_items": [
    {
      "sr_no": 1,
      "hsn_sac_code": "8471",
      "description_of_services": "BEING LAPTOP TRF TO HINDALCO- ASWANI TIBRAWAL",
      "period": null,
      "user_qty": "-",
      "measure_of_unit": "-",
      "rate": "-",
      "amount_rs": 9031.36
    },
    {
      "sr_no": 2,
      "hsn_sac_code": "8471",
      "description_of_services": "BEING LAPTOP TRF TO HINDALCO -SANJIT DAS",
      "period": null,
      "user_qty": "-",
      "measure_of_unit": "-",
      "rate": "-",
      "amount_rs": 9031.36
    },
    {
      "sr_no": 3,
      "hsn_sac_code": "8471",
      "description_of_services": "BEING LAPTOP TRF TO HINDALCO- SALMAN HUSSAIN",
      "period": null,
      "user_qty": "-",
      "measure_of_unit": "-",
      "rate": "-",
      "amount_rs": 39502.54
    }
  ],
  "summary": {
    "total_amount": 57565.26,
    "cgst": {
      "rate": "9.00%",
      "amount": 5180.87
    },
    "sgst": {
      "rate": "9.00%",
      "amount": 5180.87
    },
    "grand_total_rs": 67927.00,
    "grand_total_in_words": "SIXTY SEVEN THOUSAND NINE HUNDRED TWENTY SEVEN RUPEES ONLY."
  },
  "tax_details": {
    "reverse_charge_applicable": "No"
  },
  "payment_details": {
    "bank_name": "HDFC BANK LIMTED",
    "account_no": "05012320000105",
    "ifsc_code": "HDFC0000501",
    "payment_inquiry_emails": [
      "p.chirania@adityabirla.com",
      "sandesh.hadkar@adityabirla.com"
    ]
  },
  "authorized_signatory": {
    "name": "RAJIV GUPTA",
    "digital_signature_date": "2020.09.22",
    "digital_signature_time": "15:57:27 +05:30"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_14.png ===


```json
{
  "invoice_details": {
    "invoice_number": "873100",
    "date_of_issue": "22/04/2020",
    "due_date": "22/05/2020",
    "currency": "INR",
    "amount_due": 89600.00
  },
  "seller_information": {
    "company_name": "ExpoHub",
    "contact_number": "123456789",
    "address": "expohub street, ABC, 123456, India",
    "logo_text": "UIPATH EXPOHUB"
  },
  "billed_to": {
    "customer_name": "Rahul Jignesh"
  },
  "line_items": [
    {
      "description": "Shared Hosting",
      "rate": 88900.00,
      "quantity": 1,
      "line_total": 88900.00
    },
    {
      "description": "LateFee",
      "rate": 700.00,
      "quantity": 1,
      "line_total": 700.00
    },
    {
      "description": null,
      "rate": 0.00,
      "quantity": 1,
      "line_total": 0.00
    }
  ],
  "summary": {
    "subtotal": 89600.00,
    "tax": 0.00,
    "total": 89600.00,
    "amount_paid": 0.00,
    "amount_due": 89600.00
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_15.png ===


```json
{
  "invoice_header": {
    "company_name": "ExpoHub",
    "address": "expohub street, ABC, 123456, India",
    "phone_number": "123456789"
  },
  "billing_details": {
    "billed_to": "Rohan Verma",
    "date_of_issue": "22/04/2020",
    "due_date": "22/05/2020",
    "invoice_number": "873103",
    "amount_due_inr": "79,250.00"
  },
  "line_items": [
    {
      "description": "Shared Hosting",
      "rate": "78,000.00",
      "quantity": 1,
      "line_total": "78,000.00"
    },
    {
      "description": "LateFee",
      "rate": "1,250.00",
      "quantity": 1,
      "line_total": "1,250.00"
    },
    {
      "description": null,
      "rate": "0.00",
      "quantity": 1,
      "line_total": "0.00"
    }
  ],
  "summary": {
    "subtotal": "79,250.00",
    "tax": "0.00",
    "total": "79,250.00",
    "amount_paid": "0.00",
    "amount_due_inr": "79,250.00"
  }
}
```


=== Results for /home/manish/Desktop/projects/etfp/images/train/image_16.png ===


```json
{
  "invoice_details": {
    "invoice_number": "873104",
    "date_of_issue": "22/04/2020",
    "due_date": "22/05/2020",
    "currency": "INR",
    "amount_due": "48780.00"
  },
  "seller_information": {
    "name": "ExpoHub",
    "address": "expohub street, ABC, 123456, India",
    "phone_number": "123456789"
  },
  "billed_to": {
    "name": "Joydeep Sahoo"
  },
  "line_items": [
    {
      "description": "Shared Hosting",
      "rate": "48000.00",
      "quantity": 1,
      "line_total": "48000.00"
    },
    {
      "description": "LateFee",
      "rate": "780.00",
      "quantity": 1,
      "line_total": "780.00"
    },
    {
      "description": "",
      "rate": "0.00",
      "quantity": 1,
      "line_total": "0.00"
    }
  ],
  "summary": {
    "subtotal": "48780.00",
    "tax": "0.00",
    "total": "48780.00",
    "amount_paid": "0.00",
    "amount_due_final": "48780.00"
  }
}
```